contracts/src/tests/test_erc677.cairo

```diff
@@ -0,0 +1,115 @@
+use starknet::ContractAddress;
+use starknet::contract_address_const;
+use starknet::testing::set_caller_address;
+use starknet::syscalls::deploy_syscall;
+use starknet::class_hash::Felt252TryIntoClassHash;
+
+use array::ArrayTrait;
+use traits::Into;
+use traits::TryInto;
+use zeroable::Zeroable;
+use option::OptionTrait;
+use core::result::ResultTrait;
+
+use chainlink::token::mock::valid_erc667_receiver::ValidReceiver;
+use chainlink::token::mock::invalid_erc667_receiver::InvalidReceiver;
+use chainlink::libraries::token::erc677::ERC677;
+
+#[abi]
+trait MockValidReceiver {
+    fn on_token_transfer(sender: ContractAddress, value: u256, data: Array<felt252>);
+    fn supports_interface(interface_id: u32) -> bool;
+    fn verify() -> ContractAddress;
+}
+
+#[abi]
+trait MockInvalidReceiver {
+    fn supports_interface(interface_id: u32) -> bool;
+    fn set_supports(value: bool);
+}
+
+
+// Ignored tests are dependent on upgrading our version of cairo to include this PR https://github.com/starkware-libs/cairo/pull/2912/files
```

Comment:
I think we can remove this comment because there are no more ignored tests in this file